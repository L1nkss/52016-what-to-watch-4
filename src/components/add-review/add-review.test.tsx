import * as React from "react";
import * as renderer from "react-test-renderer";
import AddReview from "@components/add-review/add-review";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {createMemoryHistory} from "history";
import {Router} from "react-router";

const mockStore = configureStore([]);
const history = createMemoryHistory(`/sign-in`);

const film = {
  id: 1,
  backgroundImage: 'https://test.ru',
  name: 'Film',
  posterImage: 'https://image.com'
};

describe(`Testing AddReview component`, () => {
  it(`Component should successfully rendered`, () => {
    const store = mockStore({
      GENRE: {
        genre: `All genres`
      },
      DATA: {
        films: [],
        loading: false
      },
      PROMO: {
        film: {},
        loading: false
      },
      USER: {
        authorizationStatus: `NO_AUTH`
      }
    });
    const component = renderer.create(
      <Router history={history}>
      <Provider store={store}>
      <AddReview details={film} isError={false} isLoading={false} onSubmit={() => {}} />
    </Provider>
      </Router>
        ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
