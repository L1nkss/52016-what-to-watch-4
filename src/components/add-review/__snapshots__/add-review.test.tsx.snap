// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddReview component Component should successfully rendered 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Film"
        src="https://test.ru"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <Memo(Connect(Component))>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <Link
              className="breadcrumbs__link"
              to="/film-detail/1"
            >
              Film
            </Link>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
    </Memo(Connect(Component))>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="Film poster"
        height="327"
        src="https://image.com"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
        >
          <React.Fragment>
            <input
              className="rating__input"
              defaultChecked={false}
              disabled={false}
              id="star-1"
              name="rating"
              type="radio"
              value={1}
            />
            <label
              className="rating__label"
              htmlFor="star-1"
            >
              Rating 1
            </label>
          </React.Fragment>
          <React.Fragment>
            <input
              className="rating__input"
              defaultChecked={false}
              disabled={false}
              id="star-2"
              name="rating"
              type="radio"
              value={2}
            />
            <label
              className="rating__label"
              htmlFor="star-2"
            >
              Rating 2
            </label>
          </React.Fragment>
          <React.Fragment>
            <input
              className="rating__input"
              defaultChecked={true}
              disabled={false}
              id="star-3"
              name="rating"
              type="radio"
              value={3}
            />
            <label
              className="rating__label"
              htmlFor="star-3"
            >
              Rating 3
            </label>
          </React.Fragment>
          <React.Fragment>
            <input
              className="rating__input"
              defaultChecked={false}
              disabled={false}
              id="star-4"
              name="rating"
              type="radio"
              value={4}
            />
            <label
              className="rating__label"
              htmlFor="star-4"
            >
              Rating 4
            </label>
          </React.Fragment>
          <React.Fragment>
            <input
              className="rating__input"
              defaultChecked={false}
              disabled={false}
              id="star-5"
              name="rating"
              type="radio"
              value={5}
            />
            <label
              className="rating__label"
              htmlFor="star-5"
            >
              Rating 5
            </label>
          </React.Fragment>
        </div>
      </div>
      <div
        className="add-review__text"
      >
        <textarea
          className="add-review__textarea"
          disabled={false}
          id="review-text"
          maxLength={400}
          minLength={50}
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
  </div>
</section>
`;
